src/ring_theory/subring.lean

```diff
@@ -26,13 +26,14 @@ namespace is_ring_hom
 instance {S : set R} [is_subring S] : is_ring_hom (@subtype.val R S) :=
 by refine {..} ; intros ; refl
 
+instance is_subring_preimage {R : Type u} {S : Type v} [ring R] [ring S]
+  (f : R → S) [is_ring_hom f] (s : set S) [is_subring s] : is_subring (f ⁻¹' s) := {}
```

Comment:
What is this empty brace thing?!

Reply:
You don't have to write down the fields of the structure which have instance brackets. In this case, the fields are `is_add_subgroup` and `is_submonoid`, and type class inference takes care of both.